### PR TITLE
feat: add an option to omit variables content in the variables validator error messages

### PR DESCRIPTION
**Pull request by Sam-tesouro**
per discussion at https://github.com/wundergraph/cosmo/issues/1182

### CODE REVIEW

File: v2/pkg/variablesvalidation/variablesvalidation.go [hunk 1]
```diff
@@ -87,6 +89,11 @@ const (
 	pathItemKindArray
 )
 
+func devModeCheck() bool {
+	d := strings.ToLower(os.Getenv("DEV_MODE"))
```
Comment (jensneuse) [inline v2/pkg/variablesvalidation/variablesvalidation.go]: Can we turn this into a property on the variablesVisitor that we can pass on creation?
This way, it won't depend on env vars, etc. which makes it more universally usable.

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: v2/pkg/variablesvalidation/variablesvalidation.go (hunk 1 above)
Comment (Tesouro-Chris) [reply v2/pkg/variablesvalidation/variablesvalidation.go]: Hey @jensneuse , thanks for the suggested change. @Sam-tesouro and I were just chatting, and I'm afraid my go foo is a bit rusty so wondering how we'd pass that into the variables visitor. If you have a quick example, suggestion, or link, that'd help. We're working on it either way though. Thanks for all you do!

Comment (Sam-tesouro) [reply v2/pkg/variablesvalidation/variablesvalidation.go]: Implemented a thing that might be what you want, possibly.

Comment (Sam-tesouro) [reply v2/pkg/variablesvalidation/variablesvalidation.go]: I see the issue with being so coupled to the DEV_MODE env var, working on decoupling still. Biggest hangup currently is just figuring out how I am going to achieve test coverage.

Comment (jensneuse) [reply v2/pkg/variablesvalidation/variablesvalidation.go]: I can help you, need to see when I find the time.

Comment (Sam-tesouro) [reply v2/pkg/variablesvalidation/variablesvalidation.go]: Thank you! I believe I achieved decoupling. Glad to make further modifications if you have preferences!

**Review by devsergiy — APPROVED** [review]

**Comment by Sam-tesouro** [comment]
@devsergiy @StarpTech @jensneuse many thanks for taking this over the finish line, I was out on paternity and y'all are amazing.

**Pull request merged**